client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/channel/failover/DefaultFailoverManagerTest.java

```diff
@@ -44,13 +46,15 @@
     private Map<ServerType, DefaultFailoverManager.AccessPointIdResolution> resolutionProgressMap;
     private DefaultFailoverManager failoverManager;
     private ExecutorContext context;
+    private FailoverStrategy failoverStrategy;
 
     @Before
     public void setUp() {
         channelManager = Mockito.mock(KaaChannelManager.class);
         context = Mockito.mock(ExecutorContext.class);
         Mockito.when(context.getScheduledExecutor()).thenReturn(Executors.newScheduledThreadPool(1));
-        failoverManager = new DefaultFailoverManager(channelManager, context, RESOLUTION_TIMEOUT_MS, BOOTSTRAP_RETRY_PERIOD, 1, 1, TimeUnit.MILLISECONDS);
+        failoverStrategy = Mockito.spy(new DefaultFailoverStrategy(BOOTSTRAP_RETRY_PERIOD, 1, 1, TimeUnit.MILLISECONDS));
```

Comment:
Added ability to switch failover strategy at runtime: https://github.com/kaaproject/kaa/pull/565/commits/a7bbbc993861ec20e568968ad56a6be21c3e0dcd